### PR TITLE
Moment select

### PR DESCRIPTION
**Pull request by wytrych**
Changes the duration-select to a moment-select so it can be used instead of the timepicker.

### CODE REVIEW

**Comment by wytrych** [comment]
@incuna/js @maxpeterson Please review

File: src/scripts/directives.js-e [hunk 1]
```diff
@@ -1,295 +0,0 @@
...
-                    scope.$watch('data.model', function (value) {
-                        if (angular.isDefined(value)) {
-                            scope.model = value;
-                        }
-                    });
-                }
-            };
-        }
-    ]);
-}(window.angular, window._));
```
Comment (wytrych) [inline src/scripts/directives.js-e]: This was a leftover file after a cleanup some while ago.

File: src/scripts/directives.js [hunk 2]
```diff
@@ -325,19 +327,30 @@
                         scope.hourChoices = _.range(scope.hourMin, scope.hourMax, scope.hourStep);
                         scope.minuteChoices = _.range(scope.minuteMin, scope.minuteMax, scope.minuteStep);
                         var group = ['hours', 'minutes'];
+
+                        var cast = function (value) {
```
Comment (maxpeterson) [inline src/scripts/directives.js]: Can this be `cast =  scope.type === 'duration' ? moment.duration : moment;`

File: src/scripts/directives.js [hunk 3]
```diff
@@ -317,7 +318,8 @@
                         hourStep: '1',
                         minuteMin: '0',
                         minuteMax: '60',
-                        minuteStep: '1'
+                        minuteStep: '1',
+                        type: 'date'
```
Comment (maxpeterson) [inline src/scripts/directives.js]: It seems odd to give this a default that is not used.

Comment (maxpeterson) [inline src/scripts/directives.js]: Could it be a boolean? `useDuration: false`?

File: src/scripts/directives.js [hunk 4]
```diff
@@ -325,19 +327,30 @@
...
                                     return [key, scope[key] || 0];
                                 }));
-                                scope.model = moment.duration(values);
+
+                                scope.model = cast(values);
+
                             }
                         }, true);
                         scope.$watch('model', function (value) {
+                            value = cast(value);
```
Comment (maxpeterson) [inline src/scripts/directives.js]: Shouldn't this be inside the `if (angular.isDefined(value)) {`? I would expect `cast(undefined)` may fail or return something that is `defined`.

**Comment by wytrych** [comment]
@maxpeterson Updated

**Comment by wytrych** [comment]
@maxpeterson Any more comments?

File: src/scripts/directives.js [hunk 5]
```diff
@@ -325,19 +327,24 @@
                         scope.hourChoices = _.range(scope.hourMin, scope.hourMax, scope.hourStep);
                         scope.minuteChoices = _.range(scope.minuteMin, scope.minuteMax, scope.minuteStep);
                         var group = ['hours', 'minutes'];
+
+                        var cast = (scope.useDuration === 'true') ? moment.duration : moment;
```
Comment (maxpeterson) [inline src/scripts/directives.js]: It seems wrong to check for the string `true`, could it be cast to a boolean `JSON.parse(scope.useDuration)`?

**Comment by wytrych** [comment]
@maxpeterson updated

**Pull request merged**